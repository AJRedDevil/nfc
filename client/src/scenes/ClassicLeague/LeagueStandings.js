// npm packages
import React from 'react';

// our packages
import Animation from '../../components/animations';
import Table from '../../components/Table';
import LeagueStandingsPropTypes from './PropTypes';

export const getBody = standings =>
  standings.map(item => [
    item.rank,
    item.entry_name,
    item.player_name,
    item.event_total,
    item.total,
  ]);

const LeagueStandings = ({head, standings}) => (
  <Animation.Fade in>
    <div>
      <Table body={getBody(standings)} head={head} />
    </div>
  </Animation.Fade>
);
LeagueStandings.propTypes = {
  head: LeagueStandingsPropTypes.head,
  standings: LeagueStandingsPropTypes.body,
};

export default LeagueStandings;
