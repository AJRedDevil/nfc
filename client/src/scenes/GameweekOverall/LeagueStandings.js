// npm packages
import React from 'react';

// our packages
import Animation from '../../components/animations';
import Table from '../../components/Table';
import LeagueStandingsPropTypes from './PropTypes';

const LeagueStandings = ({head, standings}) => (
  <Animation.Fade in>
    <div>
      <Table body={standings} head={head} />
    </div>
  </Animation.Fade>
);
LeagueStandings.propTypes = {
  head: LeagueStandingsPropTypes.head,
  standings: LeagueStandingsPropTypes.body,
};

export default LeagueStandings;
