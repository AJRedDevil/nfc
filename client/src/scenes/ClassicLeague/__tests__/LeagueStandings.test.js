// npm packages
import React from 'react';
import renderer from 'react-test-renderer';

// our packages
import LeagueStandings, {getBody} from '../LeagueStandings';

const standings = [
  {
    rank: 1,
    entry_name: 'test',
    player_name: 'test_1',
    event_total: 50,
    total: 100,
  },
  {
    rank: 2,
    entry_name: 'test',
    player_name: 'test_2',
    event_total: 80,
    total: 200,
  },
];

describe('# LeagueStandings', () => {
  test('get correct body from standings', () => {
    const response = getBody(standings);
    expect(response).toEqual([
      [1, 'test', 'test_1', 50, 100],
      [2, 'test', 'test_2', 80, 200],
    ]);
  });

  test('LeagueStandings wrapper', () => {
    const head = [
      {text: 'Rank'},
      {text: 'Team'},
      {text: 'Player'},
      {text: 'GW'},
      {text: 'Total'},
    ];
    const wrapper = renderer
      .create(<LeagueStandings head={head} standings={standings} />)
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
