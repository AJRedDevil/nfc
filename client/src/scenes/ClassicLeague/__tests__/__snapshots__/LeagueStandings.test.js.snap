// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# LeagueStandings LeagueStandings wrapper 1`] = `
<div>
  <table
    className="table is-fullwidth is-striped is-narrow is-hoverable"
  >
    <thead>
      <tr>
        <th>
          Rank
        </th>
        <th>
          Team
        </th>
        <th>
          Player
        </th>
        <th>
          GW
        </th>
        <th>
          Total
        </th>
      </tr>
    </thead>
    <tfoot>
      <tr />
    </tfoot>
    <tbody>
      <tr>
        <th>
          1
        </th>
        <td>
          test
        </td>
        <td>
          test_1
        </td>
        <td>
          50
        </td>
        <td>
          100
        </td>
      </tr>
      <tr>
        <th>
          2
        </th>
        <td>
          test
        </td>
        <td>
          test_2
        </td>
        <td>
          80
        </td>
        <td>
          200
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
